token/cli/src/main.rs

```diff
@@ -2583,23 +2597,34 @@ fn main() -> Result<(), Error> {
         );
         let websocket_url = solana_cli_config::Config::compute_websocket_url(&json_rpc_url);
 
-        let (signer, fee_payer) = signer_from_path(
-            matches,
-            matches
-                .value_of("fee_payer")
-                .unwrap_or(&cli_config.keypair_path),
-            "fee_payer",
-            &mut wallet_manager,
-        )
-        .map(|s| {
-            let p = s.pubkey();
-            (s, p)
-        })
-        .unwrap_or_else(|e| {
-            eprintln!("error: {}", e);
-            exit(1);
-        });
-        bulk_signers.push(signer);
+        // If the fee_payer is specified, or if the default is used and the file exists, then load it
+        let keypair_path = {
+            if let Some(fee_payer) = matches.value_of("fee_payer") {
+                Some(fee_payer.to_string())
+            } else if std::fs::metadata(&cli_config.keypair_path).is_ok() {
+                Some(cli_config.keypair_path.clone())
+            } else {
+                None
+            }
+        };
```

Comment:
nit: to reduce branches again (and rename the variable)

```
let fee_payer_path = matches.value_of("fee_payer")
    .map(|s| s.to_string())
    .or_else(|| std::fs::metadata(&cli_config.keypair_path).is_ok().then(|| cli_config.keypair_path.clone()));
```

Reply:
Again I find that to be more or less unreadable because it wraps control flow up in very non obvious ways, but like I said this is Solana Labs code so I will do it however you like.

Reply:
Deleting my comment, on re-read this is actually pretty easy to follow